lib/Macros/Sources/SwiftMacros/SwiftifyImportMacro.swift

```diff
@@ -204,13 +204,42 @@ func replaceBaseType(_ type: TypeSyntax, _ base: TypeSyntax) -> TypeSyntax {
   return base
 }
 
+// C++ type qualifiers, `const T` and `volatile T`, are encoded as fake generic
+// types, `__cxxConst<T>` and `__cxxVolatile<T>` respectively. Remove those.
+func dropQualifierGenerics(_ type: TypeSyntax) -> TypeSyntax {
+  guard let identifier = type.as(IdentifierTypeSyntax.self) else { return type }
+  guard let generic = identifier.genericArgumentClause else { return type }
+  guard let genericArg = generic.arguments.first else { return type }
+  guard case .type(let argType) = genericArg.argument else { return type }
+  switch identifier.name.text {
+  case "__cxxConst", "__cxxVolatile":
+    return dropQualifierGenerics(argType)
+  default:
+    return type
+  }
+}
+
+// The `const` type qualifier used to be encoded as a `_const` suffix on type
+// names (though this causes issues for more complex types). We still drop the
+// suffix here for backwards compatibility with older textual interfaces.
+func dropQualifierSuffix(_ type: TypeSyntax) -> TypeSyntax {
+  guard let identifier = type.as(IdentifierTypeSyntax.self) else { return type }
+  let typename = identifier.name.text
+  if typename.hasSuffix("_const") {
+    return TypeSyntax(identifier.with(\.name, TokenSyntax.identifier(
+      String(typename.dropLast("_const".count))
+    )))
+  }
+  return type
+}
+
 // The generated type names for template instantiations sometimes contain
-// a `_const` suffix for diambiguation purposes. We need to remove that.
-func dropConstSuffix(_ typeName: String) -> String {
-  if typeName.hasSuffix("_const") {
-    return String(typeName.dropLast("_const".count))
+// encoded qualifiers for disambiguation purposes. We need to remove those.
+func dropCxxQualifiers(_ type: TypeSyntax) -> TypeSyntax {
+  if let attributed = type.as(AttributedTypeSyntax.self) {
```

Comment:
I am not sure I follow why we branch on `AttributedTypeSyntax`.  My understanding is that we get this when we have `borrowing` or similar specifiers in front of the type: https://swiftpackageindex.com/swiftlang/swift-syntax/600.0.1/documentation/swiftsyntax/simpletypespecifiersyntax

If we want to remain compatible with the older strings, I think the only thing we can do is to do both handling every time.

Reply:
Ah, I see! My bad, I did not notice doing both below. Shouldn't we also `dropQualifierSuffix` in both branches?

Reply:
This is a recursive call to `dropCxxQualifiers`; `dropQualifierSuffix` and `dropQualifierGenerics` are only called after this branch (:

Reply:
Wow. I should have payed more attention. Sorry, this looks good to me.